File: BUEN_SABOR_BACKEND/src/main/java/com/lab4/buen_sabor_backend/model/Articulo.java
package com.lab4.buen_sabor_backend.model;

import com.fasterxml.jackson.annotation.*;
import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;
import java.util.HashSet;
import java.util.Set;

@Entity
@Inheritance(strategy = InheritanceType.JOINED)
@DiscriminatorColumn(name = "tipo_articulo")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Articulo extends Master {

    private String denominacion;
    private Double precioVenta;


    @OneToMany(mappedBy = "articulo", cascade = {CascadeType.MERGE, CascadeType.PERSIST}, orphanRemoval = true)
    private Set<ImagenArticulo> imagenes;

    @ManyToOne
    @JoinColumn(name = "unidad_medida_id")
    private UnidadMedida unidadMedida;

    @OneToMany(mappedBy = "articulo", cascade = {CascadeType.MERGE, CascadeType.PERSIST})
    private List<DetallePromocion> detallesPromocion = new ArrayList<>();

    @ManyToOne
    @JoinColumn(name = "categoria_id")
    private Categoria categoria;

}

File: BUEN_SABOR_FRONTEND/src/models/Articulo.ts
import ImagenArticulo from "./ImagenArticulo";
import UnidadMedida from "./UnidadMedida";
import Categoria from "./Categoria";
import type TipoArticulo from "./enums/TipoArticulo";

export default class Articulo {
    id?: number;
    denominacion: string = "";
    precioVenta?: number | null;
    imagenes: ImagenArticulo[] = [];
    unidadMedida!: UnidadMedida;
    categoria!: Categoria;
    tipoArticulo!: TipoArticulo;
    eliminado!: boolean;
}